### ISSUE TITLE
Single reference is generated as "multiple" (array)

### ISSUE THREAD

**Issue by Cry0nicS** [Contentstack-Solutions/contentstack-cli-tsgen#43]
A referenced field that is not set as "multiple" has the type generated as an array.

Example from the demo data (from the starters): a header object has a reference to a page - `page_reference`. It looks like this
```json
  {
      "data_type": "reference",
      "display_name": "Page Reference",
      "reference_to": [
          "page"
      ],
      "field_metadata": {
          "ref_multiple": false,
          "ref_multiple_content_types": true,
          "description": "Refers a linked entity of type page",
          "instruction": "Refers a linked entity of type page"
      },
      "uid": "page_reference",
      "non_localizable": false,
      "mandatory": false,
      "multiple": false,  
      "unique": false
  }
```

However, the generated type looks like this:
```ts
    navigation_menu?: [
        {
            /** Label */
            label?: string;
            /** Page Reference */
            page_reference?: Page[]; // <--- should not be an array unless "multiple" is set to "true"
        },
    ]
```

I tried changing it to multiple and back to single, but nothing changes with regards to the generated type.

Version:
```
❯ csdx plugins
contentstack-cli-tsgen 2.1.6
```

**Comment by netrajpatel** [comment]
Hi @Cry0nicS,
We are looking into this and we will get back to you soon with updates.
Thanks!

**Comment by aman19K** [comment]
Hi @Cry0nicS , We have handled this now. Install the latest TSGEN v2.1.8 plugin to get the fix.

**Comment by sgoess** [comment]
This change causes some new TS issues in our existing code base. For example, I have a Content Model with two reference fields. One is for a single reference, one is for multiple references. This now with v.2.1.8 generates the following types:

```TSX
export interface TestSg {
  title: string;
  referenceSingle?: Banner;
  referenceMultiple?: Banner[];
}
```

However, both types are actually returned as arrays from the API:

```JSON
{
    "reference_multiple": [
        {
            "uid": "blt5eb659f9251c84f6",
            "_content_type_uid": "banner"
        },
        {
            "uid": "bltac62e4a814af4dcb",
            "_content_type_uid": "banner"
        }
    ],
    "reference_single": [
        {
            "uid": "blt5eb659f9251c84f6",
            "_content_type_uid": "banner"
        }
    ],
    "title": "Test TSGen"
}
```

IMO this fix for v2.1.8 needs a rollback to have the API output match the types again.
